src/lib.rs

```diff
@@ -796,7 +796,24 @@ where
 //
 // ****************************************************************************
 
-// None
+/// Transform mode variants (ReadWriteCreate_Or_Append) to simple modes ReadWriteAppend
```

Comment:
Can you explain the reason behind this logic?

Reply:
I'm confused by your question. This is just the lines 487-500 from the old file extracted in a separate function. What is missing?

Reply:
Sorry, I was just reading this function and I couldn't work out what it was trying to do or why!